app/coffeescript/components/mixins/with_default_attributes.coffee

```diff
@@ -5,19 +5,29 @@ define [
 ], (
   $
   compose
-  mapUtils
+  flightUtils
 ) ->
 
   withDefaultAttr = ->
 
     @defaultAttrs = (properties) ->
-      flight.utils.push(@defaults, properties, true) || (@defaults = properties)
+      flightUtils.push(@defaults, properties, true) || (@defaults = properties)
 
     @initAttributes = (attrs = {}) ->
       attr = Object.create(attrs)
       attr[k] = v for k,v of this.defaults when (!attrs.hasOwnProperty(k))
       this.attr = attr
 
-    return
-    
+    # http://andrewdupont.net/2009/08/28/deep-extending-objects-in-javascript/
+    @mergeAttributes = (destination, source) ->
+      console.log "in mergeAttributes"
+      for property of source
+        console.log "  property: #{property}"
+        if (source[property] && source[property].constructor && source[property].constructor == Object)
+          destination[property] = destination[property] || {}
+          console.log "    recurse: #{property}"
+          arguments.callee(destination[property], source[property])
+        else
+          destination[property] = source[property]
+
   return withDefaultAttr
```

Comment:
We should not clobber. If we need a deep merge let's implement it using flightjs utils.push (better code). If we need to clobber let us have 2 functions.